src/elements/walletConnect/WalletConnectButton.tsx

```diff
@@ -1,26 +1,26 @@
 import { UseWalletConnect } from './useWalletConnect';
-import { shortenString } from 'utils/pureFunctions';
-import { Image } from 'components/image/Image';
+import { shortenString, truncateString } from 'utils/pureFunctions';
 import { ReactComponent as IconWallet } from 'assets/icons/wallet.svg';
+import useENS from 'hooks/useENS';
+import Davatar from '@davatar/react';
 
 export const WalletConnectButton = ({
   handleWalletButtonClick,
   account,
   selectedWallet,
 }: UseWalletConnect) => {
-  const buttonText = account ? shortenString(account) : 'Connect Wallet';
+  const { ensName } = useENS(account || '');
+  const buttonText = account
+    ? truncateString(ensName || '') || shortenString(account)
```

Comment:
I'd expect not having to do this `ensName || '' `when passing a param - but truncateString should handle it for me

Reply:
ideally we use the existing `shortenString` helper for this.

Reply:
@pingustar `shortenString` add separator to the middle of the text and it acts weirdly with big names and fixed length 😕

Reply:
hmm maybe we can change it to something more generic and add the option to add the placeholder in the middle or at the end.

Reply:
@pingustar I think its better to do in follow up PR and make this one simple 🤔  WDYT?